### PR TITLE
Allow BTC In Crowdsales

### PR DESCRIPTION
**Pull request by zathras-crypto**
This PR enables the use of Bitcoin as a funding source in crowdsales.

This is currently a WIP/placeholder, but should be functional.

Current test results:

```
Preparing a test environment...
   * Starting a fresh regtest daemon
   * Preparing some mature testnet BTC
   * Obtaining a master address to work with
   * Funding the address with some testnet BTC for fees
   * Participating in the Exodus crowdsale to obtain some OMNI
   * Generating addresses to use

Testing a crowdsale using BTC before activation...
   * Creating an indivisible test property and opening a crowdsale
     # Checking the transaction was invalid... PASS

Activating v2 crowdsales (allow BTC)... 
   * Sending the activation
     # Checking the activation transaction was valid... PASS
   * Mining 10 blocks to forward past the activation block
     # Checking the activation went live as expected... PASS

Testing crowdsales using BTC after activation...

Testing an indivisible BTC crowdsale...
   * Creating an indivisible test property and opening a crowdsale
     # Checking the transaction was valid... PASS

Testing sending a BTC payment to the crowdsale...
   * Sending some BTC to mkxYY4KGgyBKgsXqgjEj75bsh4yxd2Kx1n
   * Sending some BTC from mkxYY4KGgyBKgsXqgjEj75bsh4yxd2Kx1n to the crowdsale
     # Checking the transaction was valid... PASS
     # Checking the payment was linked to the correct transaction... PASS
     # Checking the payment was linked to the correct recipient... PASS
     # Checking the payment amount was 0.1 BTC... PASS
     # Checking the sending address now has 100 of property 3... PASS
     # Checking the transaction is listed in crowdsale participants... PASS
     # Checking the crowdsale amount raised is now 0.1 BTC... PASS
     # Checking the amount of tokens issued to users is now 100... PASS
     # Checking the amount of tokens issued to uthe isuser is still 0... PASS

Testing a divisible BTC crowdsale...
   * Sending some BTC to mz41y8JmRuU3JujMx5cua2rHATsy3nfsj9
   * Sending some BTC to msbm3v3HUtJzpL9u5iZ2uGUVv5CPySAzqF
   * Creating a divisible test property and opening a crowdsale
     # Checking the transaction was valid... PASS
   * Sending some BTC from msbm3v3HUtJzpL9u5iZ2uGUVv5CPySAzqF to the crowdsale
     # Checking the transaction was valid... PASS
     # Checking the payment was linked to the correct transaction... PASS
     # Checking the payment was linked to the correct recipient... PASS
     # Checking the payment amount was 0.1 BTC... PASS
     # Checking the sending address now has 100.0 of property 4... PASS
     # Checking the transaction is listed in crowdsale participants... PASS
     # Checking the crowdsale amount raised is now 0.1 BTC... PASS
     # Checking the amount of tokens issued to users is now 100.0... PASS
     # Checking the amount of tokens issued to uthe isuser is still 0... PASS

####################
#  Summary:        #
#    Passed = 23   #
#    Failed = 0    #
####################
```

### CODE REVIEW

**Comment by dexX7** [comment]
Hey @zathras-crypto, thanks for pushing this!

I'm a bit surprised by the title: are you suggesting we should adopt this for 0.0.11?

**Comment by zathras-crypto** [comment]
Sorry my bad, that came from my branch name - nope not intended for 0.0.11.  I've fixed the title now :)

File: src/omnicore/createpayload.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #include "omnicore/createpayload.h"
 
 #include "omnicore/convert.h"
+#include "omnicore/omnicore.h"
```
Comment (dexX7) [inline src/omnicore/createpayload.cpp]: This probably shouldn't be here.

**Comment by dexX7** [comment]
Given this PR is outdated and needs further review, it's going to be closed for now.

**Pull request closed without merging**